blog/_posts/2024-05-14-core-extension.md

```diff
@@ -0,0 +1,87 @@
+### What Are Kubernetes Extensions?
+
+[Kubernetes extensions](https://learn.microsoft.com/en-us/azure/aks/cluster-extensions?tabs=azure-cli) (or cluster extensions) are pre-packaged applications that simplify the installation and lifecycle management of Azure capabilities on Kubernetes clusters. Examples include [Azure Backup](https://learn.microsoft.com/en-us/azure/backup/azure-kubernetes-service-backup-overview), [GitOps (Flux)](https://learn.microsoft.com/en-us/azure/azure-arc/kubernetes/conceptual-gitops-flux2), and [Azure Machine Learning](https://learn.microsoft.com/en-us/azure/machine-learning/how-to-attach-kubernetes-anywhere?view=azureml-api-2). Third-party extensions (or Kubernetes apps), such as [Datadog AKS Cluster Extension](https://azuremarketplace.microsoft.com/en-us/marketplace/apps/datadog1591740804488.dd_aks_extension?tab=Overview) and [Isovalent Cilium Enterprise](https://azuremarketplace.microsoft.com/en-us/marketplace/apps/isovalentinc1662143158090.isovalent-cilium-enterprise?tab=Overview), are also available in the Azure Marketplace.
+
+### Extensions vs. Add-ons
+
+Extensions and add-ons both enhance AKS functionality but differ in scope and management:
+
+- **Add-ons**: Part of the AKS resource provider (`Microsoft.ContainerService/managedClusters`), limited to AKS, and generally reserved for cluster critical functionality.
+- **Extensions**: Managed via the Extension API (`Microsoft.KubernetesConfiguration/extensions`), supporting both AKS and [Arc-enabled Kubernetes environments](https://learn.microsoft.com/en-us/azure/azure-arc/kubernetes/overview).
+
+### What Are Core Kubernetes Extensions?
+
+ Core Kubernetes extensions are a Kubernetes extension that provides broader region availability, a native AKS experience, safer version management, enhanced security, efficiency, and reliability on AKS. Important AKS capabilities can be supported through core Kubernetes extensions while maintaining seamless integration with AKS just like add-ons. This approach also paves the way for more services and applications to be supported across both AKS and Arc-connected Kubernetes environments in the future.
+
+#### Broader region availability
+Core Kubernetes extensions are available in all regions across Azure Public cloud, Azure Government and China regions. See the full list of regions at [Azure Arc enabled Kubernetes region support](https://azure.microsoft.com/en-us/explore/global-infrastructure/products-by-region/?products=azure-arc&regions=all).
+
+#### Native AKS experience
+* Add-ons migrated to core extensions: Some AKS add-ons, such as Container Insights, Azure Managed Prometheus, and App Monitoring, are transitioning to core extensions. Although the underlying technology shifts to extensions, the user experience remains unchanged. The CLI command to enable/disable migrated add-ons are the same as before migration.
+
+    ```bash
+    az aks create/update --resource-group <group> --name <cluster> --enable/disable <add-on>
```

Comment:
it's not going to be just '--enable/disable' params and the scenario name would also be in that parameter no? sharing an example would be good. [This](https://learn.microsoft.com/en-us/azure/defender-for-cloud/defender-for-containers-enable?tabs=k8s-deploy-cli%2Ck8s-deploy-asc%2Ck8s-verify-asc%2Ck8s-remove-arc%2Caks-removeprofile-api&pivots=defender-for-container-aks#use-the-azure-cli-to-deploy-the-defender-sensor) one has an example - 

```
az aks create --enable-defender --resource-group <your-resource-group> --name <your-cluster-name>
```